internal/functions/deploy/deploy.go

```diff
@@ -56,21 +68,82 @@ func getFunctionSlugs(fsys afero.Fs) ([]string, error) {
 	return slugs, nil
 }
 
-func bundleFunction(ctx context.Context, entrypointPath, importMapPath, buildScriptPath string) (*bytes.Buffer, error) {
-	denoPath, err := utils.GetDenoPath()
+func bundleFunction(ctx context.Context, dockerEntrypointPath, importMapPath string, fsys afero.Fs) (*bytes.Buffer, error) {
+	cwd, err := os.Getwd()
 	if err != nil {
 		return nil, err
 	}
-	// Bundle function and import_map with deno
-	args := []string{"run", "-A", buildScriptPath, entrypointPath, importMapPath}
-	cmd := exec.CommandContext(ctx, denoPath, args...)
-	var outBuf, errBuf bytes.Buffer
-	cmd.Stdout = &outBuf
-	cmd.Stderr = &errBuf
-	if err := cmd.Run(); err != nil {
-		return nil, fmt.Errorf("Error bundling function: %w\n%v", err, errBuf.String())
+
+	// create temp directory to store generated eszip
+	tmpDir, err := afero.TempDir(fsys, "eszip")
+	if err != nil {
+		return nil, err
 	}
-	return &outBuf, nil
+	defer fsys.RemoveAll(tmpDir)
+	outputPath := dockerOutputDir + "/output.eszip"
+
+	binds := []string{
+		// Reuse deno cache directory, ie. DENO_DIR, between container restarts
+		// https://denolib.gitbook.io/guide/advanced/deno_dir-code-fetch-and-cache
+		utils.EdgeRuntimeId + ":/root/.cache/deno:rw,z",
+		filepath.Join(cwd, utils.FunctionsDir) + ":" + utils.DockerFuncDirPath + ":ro,z",
+		tmpDir + ":" + dockerOutputDir + ":rw,z",
+	}
+
+	if importMapPath != "" {
+		modules, err := utils.BindImportMap(importMapPath, dockerImportMapPath, fsys)
+		if err != nil {
+			return nil, err
+		}
+		binds = append(binds, modules...)
+	}
```

Comment:
Don't we need to account for import maps for individual functions? See https://github.com/supabase/cli/blob/main/internal/functions/serve/serve.go#L232

Reply:
fallbackImportMap is already set in line 205 in `deployOne` function, which should give the same result?